compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/PersistRecordValidator.java

```diff
@@ -543,4 +567,459 @@ public void reportDiagnosticInfo(SyntaxNodeAnalysisContext ctx, Location locatio
+                    for (int i = 0; i < imports.size(); i++) {
+                        ImportDeclarationNode importDeclarationNode = imports.get(i);
+                        SeparatedNodeList<IdentifierToken> moduleNames = importDeclarationNode.moduleName();
+                        if (isNodeType(importDeclarationNode, nodeType, moduleNames)) {
+                            if (hasValidModuleName(moduleNames, symbol) &&
+                                    hasValidOrgName(importDeclarationNode, symbol)) {
+                                isUserDefinedType = true;
+                                Object[] properties = checkRelationShip(recordName, type, ctx);
+                                if (isArrayType && properties.length == 5) {
+                                    Utils.reportDiagnostic(ctx, field.location(),
+                                            DiagnosticsCodes.PERSIST_115.getCode(),
+                                            MessageFormat.format(DiagnosticsCodes.PERSIST_115.getMessage(), type),
+                                            DiagnosticsCodes.PERSIST_115.getSeverity());
+                                } else {
+                                    tableAssociationType = properties[1].toString();
+                                    hasRelationAnnotation = properties[2].toString();
+                                    referenceRecord = (TypeDefinitionNode) properties[3];
+                                    if (isArrayType) {
+                                        if (!tableAssociationType.equals(Constants.ONE_TO_ONE)) {
+                                            Utils.reportDiagnostic(ctx, field.location(),
+                                                    DiagnosticsCodes.PERSIST_114.getCode(),
+                                                    DiagnosticsCodes.PERSIST_114.getMessage(),
+                                                    DiagnosticsCodes.PERSIST_114.getSeverity());
+                                        }
+                                    }
+                                    break;
+                                }
+                            } else {
+                                if (isArrayType) {
+                                    Utils.reportDiagnostic(ctx, node.location(),
+                                            DiagnosticsCodes.PERSIST_120.getCode(),
+                                            DiagnosticsCodes.PERSIST_120.getMessage(),
+                                            DiagnosticsCodes.PERSIST_120.getSeverity());
+                                }
+                                validateType(ctx, node, qualifiedNameReferenceNode.toString().trim());
+                                break;
+                            }
+                        }
+                    }
+                } else {
+                    validateType(ctx, node, type);
+                }
+            } else if (node instanceof SimpleNameReferenceNode) {
```

Comment:
Do we have to support this? This is variable usage isnt it?

Reply:
Yes, 
This type will come here.
```
User author?;
```

Reply:
Lets rename the `node` to something more specific.